drivers/gpio/gpio_esp32.c

```diff
@@ -355,6 +428,9 @@ GPIO_DEVICE_INIT(DT_INST_1_ESPRESSIF_ESP32_GPIO_LABEL, 32_to_39);
 
 static void gpio_esp32_isr(void *param)
 {
+	/* FIXME: There is a problem in clearing interrupts
+	 *  of type LEVEL_LOW/HIGH, hence it is not usable
+	 */
```

Comment:
We should move this code from `gpio_esp32_fire_callbacks` here
```
	struct gpio_esp32_data *data = device->driver_data;
	u32_t values = *data->port.irq.status_reg;

	*data->port.irq.ack_reg = values;
```
It conceptually belongs to ISR routine, also we need to write to `ack_reg` before executing callbacks.